### PR TITLE
Fixed Device Token not sent to Customer io Immediately

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

Previously we used to send device token to customer io only on the life cycle events which were sent by the Mobile SDK's, but this is resulting in a delay from sending it over to customer io, as the next life cycle event is sent on app launch/minimize from the moment the device token was handed over to RudderStack SDK's.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/customerio/transform.js [hunk 1]
```diff
@@ -36,6 +36,16 @@ const deviceRelatedEventNames = [
   "Application Opened",
   "Application Uninstalled"
```
Comment (sanpj2292) [inline v0/destinations/customerio/transform.js:37]: No new event added for `device registration` ?

File: v0/destinations/customerio/transform.js [hunk 2]
```diff
@@ -36,6 +36,16 @@ const deviceRelatedEventNames = [
...
 ];
+
+const isdeviceRelatedEventName = (eventName, destination) => {
+  return (
+    deviceRelatedEventNames.includes(eventName) ||
+    (destination.Config &&
+      destination.Config.deviceTokenEventName &&
+      destination.Config.deviceTokenEventName === eventName)
```
Comment (sanpj2292) [inline v0/destinations/customerio/transform.js:45]: I think this is different from what was decided.
Basically my perspective of the decision made doesn't involve the destination's configuration(I mean UI), I was thinking more from a perspective that an event(Device Registration) is triggered from the user's side after we get the device token.

Let me know if there is some mis-understanding here

**Review by sanpj2292 — DISMISSED** [review]
LGTM

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Add new unit test cases for the changes introduced

**Review by saikumarrs — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**